src/components/Tag/Tag.tsx

```diff
@@ -99,7 +122,7 @@ export const Tag = (props: ITagProps): React.ReactElement => {
 		>
 			{hasOtherChildren && (
 				<span className={cx('&-other-children')}>
-					{otherChildren}
+					<span onClick={handleClick}>{otherChildren}</span>
 					{isRemovable && (
 						<CloseIcon
 							onClick={handleRemove}
```

Comment:
I think the design of this component is to handle the onClick event just with the CloseIcon. So you set isRemovable to true and handle the click event in the "handleRemove" handler.